### PR TITLE
Replace send-return with a custom error throw

### PR DESCRIPTION
**Pull request by Jotra04**
Enjoy this behemoth of untested code 😉

Edit: JK, it's not that much

### CODE REVIEW

File: src/main/java/commands/roles/AddRoleAssign.java [hunk 1]
```diff
@@ -33,29 +34,25 @@ public CommandState canBeExecuted(long guildId, long channelId, Member member) {
...
-        } else if (args.length == 2) {
-            e.getChannel().sendMessage("No role and name given\n" + getUsage()).queue();
-        } else if (args.length == 3) {
-            e.getChannel().sendMessage("No name given\n" + getUsage()).queue();
-        } else if (args.length >= 4 && dataHandler.getRoleCategories(e.getGuild().getIdLong()).contains(args[0])) {
-            if (!hasEmoji(e.getMessage(), args[1])) {
-                e.getChannel().sendMessage(String.format("%s is not a valid emoji\n%s", args[1], getUsage())).queue();
-                return;
-            }
+        if (args.length == 1)
```
Comment (Zelzahn) [inline src/main/java/commands/roles/AddRoleAssign.java:37]: I believe that a switch would be more readable here.

File: src/main/java/utils/logging/MyFileAppender.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        fileAppender.setThreshold(Level.ALL);
+        fileAppender.setFile(String.format("%s/log.html", MyFileAppender.folder));
+        fileAppender.setLayout(layout);
+        fileAppender.activateOptions();
+    }
+
+
+    @Override
+    public void addFilter(Filter filter) {
+
```
Comment (Zelzahn) [inline src/main/java/utils/logging/MyFileAppender.java:29]: Add `throw new NotImplementedException("This function shouldn't be called");` to all unimplemented functions.

**Review by Zelzahn — CHANGES_REQUESTED** [review]

File: src/main/java/commands/EditSuggestion.java [hunk 3]
```diff
@@ -49,10 +50,10 @@ public void run(String[] args, GuildMessageReceivedEvent e) throws Exception{
                     m.editMessage(eb.build()).queue();
```
[the remark this answers is not in the document]
Comment (DaanWet) [reply src/main/java/commands/EditSuggestion.java:50]: Will be fixed in #75

File: src/main/java/commands/AmongUs.java [hunk 4]
```diff
@@ -30,7 +31,7 @@ public void run(String[] args, GuildMessageReceivedEvent e) throws Exception {
             //    e.getChannel().sendMessage("Wrong voicechannel noob").queue(m -> m.delete().queueAfter(10, TimeUnit.SECONDS));
             //}
         } else {
-            e.getChannel().sendMessage("You're not connected to a voice channel").queue(m -> m.delete().queueAfter(10, TimeUnit.SECONDS));
+            throw new MessageException("You're not connected to a voice channel", 10);
```
[the remark this answers is not in the document]
Comment (DaanWet) [reply src/main/java/commands/AmongUs.java:34]: Yes, part of #76

File: src/main/java/commands/pictures/DeletePicture.java [hunk 5]
```diff
@@ -47,27 +48,27 @@ public void closeExplorer(String command, Message message) {
...
-                    m.addReaction("U+25B6").queue();
-                    m.addReaction("U+274C").queue();
-                    autoClosers.put(args[0], deleteMessage(args[0], m).queueAfter(10, TimeUnit.MINUTES));
-                });
-            }
+            if (openExplorers.containsKey(args[0]))
+                throw new MessageException("An explorer is already open");
+
+            EmbedBuilder eb = new EmbedBuilder();
+            eb.setImage(String.format("http://zwervers.wettinck.be/%s/%d&%d=%d", args[0], 0, random.nextInt(), random.nextInt()));
```
[the remark this answers is not in the document]
Comment (DaanWet) [reply src/main/java/commands/pictures/DeletePicture.java:58]: Yes, part of #76

File: src/main/java/utils/logging/MyHTMLLayout.java [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.Date;
+
+public class MyHTMLLayout extends HTMLLayout {
+
+    private final DateTimeFormatter dtf = DateTimeFormatter.ofPattern("dd/MM - HH:mm:ss");
+    private final String TD = "<td>";
+    private final String CTD = "</td>";
+    private static String url = "https://pingo.wettinck.be/logs/";
```
[the remark this answers is not in the document]
Comment (DaanWet) [reply src/main/java/utils/logging/MyHTMLLayout.java:22]: Yes, part of #76

File: src/main/java/commands/roles/AddRoleAssign.java (hunk 1 above)
Comment (DaanWet) [reply src/main/java/commands/roles/AddRoleAssign.java:37]: Will fix together with #76

File: src/main/java/utils/logging/ErrorLayout.java [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    @Override
+    public boolean ignoresThrowable() {
+        return false;
+    }
+
+    @Override
+    public void activateOptions() {
+
```
Comment (Zelzahn) [inline src/main/java/utils/logging/ErrorLayout.java:44]: A NotImplementedError could be placed here

**Review by Zelzahn — APPROVED** [review]

**Pull request merged**